core/sr-primitives/src/weights.rs

```diff
@@ -74,3 +81,85 @@ impl Default for TransactionWeight {
 		TransactionWeight::Basic(0, 1)
 	}
 }
+
+/// Representation of a weight multiplier. This represents how a fee value can be computed from a
+/// weighted transaction.
+///
+/// This is basically a wrapper for the `Fixed64` type a slightly tailored multiplication to u32
+/// in the form of the `apply_to` method.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct WeightMultiplier(Fixed64);
+
+impl WeightMultiplier {
+	/// Apply the inner Fixed64 as a weight multiplier to a weight value.
+	///
+	/// This will perform a saturated  `weight + weight * self.0`.
+	pub fn apply_to(&self, weight: Weight) -> Weight {
+		let inner = self.0;
```

Comment:
we really shouldn't be mucking around with the inner member of a `Fixed64` in here and depending on `DIV` being public API (it shouldn't be). operations should really be defined purely in terms of a proper public API of `Fixed64`.